server/store/middleware/infohash/blacklist_test.go

```diff
@@ -10,41 +10,42 @@ import (
 	"github.com/stretchr/testify/assert"
 
 	"github.com/chihaya/chihaya"
-	"github.com/chihaya/chihaya/server"
 	"github.com/chihaya/chihaya/server/store"
 	"github.com/chihaya/chihaya/tracker"
-
-	_ "github.com/chihaya/chihaya/server/store/memory"
 )
 
-var srv server.Server
+type storeMock struct {
+	strings map[string]struct{}
+}
+
+func (ss *storeMock) PutString(s string) error {
+	ss.strings[s] = struct{}{}
+
+	return nil
+}
+
+func (ss *storeMock) HasString(s string) (bool, error) {
+	_, ok := ss.strings[s]
+
+	return ok, nil
+}
+
+func (ss *storeMock) RemoveString(s string) error {
+	delete(ss.strings, s)
+
+	return nil
+}
+
+var mock store.StringStore = &storeMock{
+	strings: make(map[string]struct{}),
+}
 
 func TestASetUp(t *testing.T) {
-	serverConfig := chihaya.ServerConfig{
-		Name: "store",
-		Config: store.Config{
-			Addr: "localhost:6880",
-			StringStore: store.DriverConfig{
-				Name: "memory",
-			},
-			ClientStore: store.DriverConfig{
-				Name: "memory",
-			},
-			IPStore: store.DriverConfig{
-				Name: "memory",
-			},
-			PeerStore: store.DriverConfig{
-				Name: "memory",
-			},
-		},
+	mustGetStore = func() store.StringStore {
```

Comment:
I could pull this into the test functions and get rid of the `TestASetup` function, should I?